darwin/PlatformHelpers.c

```diff
@@ -86,3 +86,36 @@ bool Platform_isRunningTranslated(void) {
    }
    return ret;
 }
+
+void Platform_calculateNanosecondsPerMachTick(uint64_t* numer, uint64_t* denom) {
+   // Check if we can determine the timebase used on this system.
+   // If the API is unavailable assume we get our timebase in nanoseconds.
+#ifdef HAVE_MACH_TIMEBASE_INFO
+   /* WORKAROUND for `mach_timebase_info` giving incorrect values on M1 under Rosetta 2.
+    *    rdar://FB9546856 https://openradar.appspot.com/radar?id=5055988478509056
```

Comment:
The URL was broken. Perhaps we can update the reference to http://www.openradar.appspot.com/FB9546856

Reply:
Fixed